app/src/main/java/com/example/recipeapp/viewmodels/LoginViewModel.java

```diff
@@ -27,11 +28,12 @@ public void login(View view) {
                 username,
                 password,
                 (user, e) -> {
+                    Context context = view.getContext();
                     if(e != null) {
                         Log.i(TAG, "Issue with login" + e);
+                        Toast.makeText(context, "Unable to login", Toast.LENGTH_SHORT).show();
                         return;
                     }
-                    Context context = view.getContext();
                     context.startActivity(new Intent(context, MainActivity.class));
```

Comment:
Same goes for these , we should observing on these inside activity https://stackoverflow.com/a/46827625